public/scripts/upload.js

```diff
@@ -34,10 +34,17 @@ async function runWorker(worker) {
       window.localforage
         .setItem("pictureArray", e.data)
         .then(function (pictureArray) {
+          if (e.data["error"] !== undefined) {
+            console.log("Get error while generating mockup", e.data["error"]);
+            window.viewModel.cancelMockup();
+            setTimeout(() => {
+              alert("Generation failed, please reload and try again later");
+            }, 100);
```

Comment:
Why need `setTimeout`?  If wanna manipulate event loop orders, can add comment to specify which event this wanna be behind of?

Reply:
Can use async/await for this?

Reply:
@YayunHuang seems still using `setTimeout`